patterns/product-hero.php

```diff
@@ -4,43 +4,43 @@
  * Slug: woocommerce-blocks/product-hero
  * Categories: WooCommerce
  */
+
+$query = new \WC_Product_Query(
+	array(
+		'limit'  => 1,
+		'return' => 'ids',
+		'status' => array( 'publish' ),
+	)
+);
+
+$products   = $query->get_products();
+$product_id = $products ? $products[0] : null;
```

Comment:
While I agree this is a valid way to provide an initial/default product for the user to work with, I believe, ideally, we should somehow display a list of available products for the user to choose from before inserting the pattern, similar to what is displayed when inserting the single product block itself in the editor:

<img width="668" alt="Screenshot 2023-06-27 at 17 32 30" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/b7ab2bd1-8cce-4485-9095-60e790452b8c">

We don't necessarily have to solve this problem on this PR, but IMO it would be a good idea to open a separate issue and explore the feasibility of something along these lines. What are your thoughts?

Reply:
Removing that PHP snippet to select a product would give us that same single product selector from the image above, but the problem then is that we lose the preview 👇 so the pattern is not really visible until inserted, making it less likely for a user to insert it.
<img width="367" alt="Screenshot 2023-06-28 at 10 24 54" src="https://github.com/woocommerce/woocommerce-blocks/assets/186112/ea4409b9-ac02-46ea-9279-74c3a22893ba">

I think eventually GB will have something to provide a custom preview, but until now, that's the only way to show the actual pattern there.

Reply:
> Removing that PHP snippet to select a product would give us that same single product selector from the image above, but the problem then is that we lose the preview 👇 so the pattern is not really visible until inserted, making it less likely for a user to insert it.

Oh ok, gotcha! Feels like we are stuck in a bad loop: without a preview, the user might not insert it, but if they do insert it and can't change the product, they may not use it either 😭 

> I think eventually GB will have something to provide a custom preview, but until now, that's the only way to show the actual pattern there.

Out of curiosity, do you know if there's an issue on GB already for allowing us to modify this custom preview?

Reply:
Opened https://github.com/woocommerce/woocommerce-blocks/issues/10022 so we can discuss this separately and unblock this work =)

Reply:
I found this issue https://github.com/WordPress/gutenberg/issues/28135 but there's not much activity, not sure if there are more.

They can actually change the product (it's still a `Single Product` block) though it may not be obvious after inserting, that's true.
<img width="683" alt="Screenshot 2023-06-28 at 12 52 59" src="https://github.com/woocommerce/woocommerce-blocks/assets/186112/2464b016-b4d6-499c-81cf-0e8947375b33">